internal/core/data/application/reading.go

```diff
@@ -526,3 +539,80 @@ func noneAutoEventSourcesByDevice(device models.Device, dic *di.Container) ([]st
 	}
 	return sources, nil
 }
+
+func processNumericReadings(isNumeric bool, readings []dtos.BaseReading) {
+	for i := range readings {
+		if isNumeric {
+			convertToNumeric(&readings[i])
+		} else {
+			convertToString(&readings[i])
+		}
+	}
+}
+
+// convertToNumeric converts SimpleReadding value to NumericReading value
+func convertToNumeric(reading *dtos.BaseReading) {
+	if len(reading.Value) == 0 {
+		return
+	}
+	switch reading.ValueType {
+	case common.ValueTypeInt8, common.ValueTypeInt16, common.ValueTypeInt32, common.ValueTypeInt64:
+		reading.NumericReading.NumericValue = cast.ToInt64(reading.Value)
```

Comment:
why do you need to cast?  it should come from the model numeric reading value directly?

Reply:
The reading value might be string from the DB.